code/modules/food_and_drinks/food/foods/sandwiches.dm

```diff
@@ -136,18 +145,171 @@
+/obj/item/reagent_containers/food/snacks/burger/crazy
+	name = "crazy hamburger"
+	desc = "This looks like the sort of food that a demented clown in a trenchcoat would make."
+	icon_state = "crazyburger"
+	filling_color = "#E3D681"
+	bitesize = 3
+	list_reagents = list("nutriment" = 10, "vitamin" = 2, "capsaicin" = 3, "condensedcapsaicin" = 2)
+	tastes = list("bun" = 2, "meat" = 4, "cheese" = 2, "beef soaked in chili" = 3, "a smoking flare" = 2)
+
+/obj/item/reagent_containers/food/snacks/burger/ppatty/white
+	name = "white pretty patty"
+	desc = "Delicious titanium!"
+	icon_state = "ppatty-mime"
+	filling_color = "#FFFFFF"
+	bitesize = 2
+	list_reagents = list("nutriment" = 7, "protein" = 1)
+	tastes = list("bun" = 2, "meat" = 2, "white" = 2)
+
+/obj/item/reagent_containers/food/snacks/burger/ppatty/red
+	name = "red pretty patty"
+	desc = "Perfect for hiding the fact that it's burnt to a crisp."
+	icon_state = "ppatty-red"
+	filling_color = "#D63C3C"
+	bitesize = 2
+	list_reagents = list("nutriment" = 7, "protein" = 1)
+	tastes = list("bun" = 2, "meat" = 2, "red" = 2)
+
+/obj/item/reagent_containers/food/snacks/burger/ppatty/orange
+	name = "orange pretty patty"
+	desc = "Contains 0% juice."
+	icon_state = "ppatty-orange"
+	filling_color = "#FFA500"
+	bitesize = 2
+	list_reagents = list("nutriment" = 7, "protein" = 1)
+	tastes = list("bun" = 2, "meat" = 2, "orange" = 2)
+
+/obj/item/reagent_containers/food/snacks/burger/ppatty/yellow
+	name = "yellow pretty patty"
+	desc = "Bright to the last bite."
+	icon_state = "ppatty-yellow"
+	filling_color = "#FFFF00"
+	bitesize = 2
+	list_reagents = list("nutriment" = 7, "protein" = 1)
+	tastes = list("bun" = 2, "meat" = 2, "yellow" = 2)
+
+/obj/item/reagent_containers/food/snacks/burger/ppatty/green
+	name = "green pretty patty"
+	desc = "It's not tainted meat, it's painted meat!"
+	icon_state = "ppatty-green"
+	filling_color = "#00FF00"
+	bitesize = 2
+	list_reagents = list("nutriment" = 7, "protein" = 1)
+	tastes = list("bun" = 2, "meat" = 2, "green" = 2)
+
+/obj/item/reagent_containers/food/snacks/burger/ppatty/blue
+	name = "blue pretty patty"
+	desc = "Is this blue rare?"
+	icon_state = "ppatty-blue"
+	filling_color = "#0000FF"
+	bitesize = 2
+	list_reagents = list("nutriment" = 7, "protein" = 1)
+	tastes = list("bun" = 2, "meat" = 2, "blue" = 2)
+
+/obj/item/reagent_containers/food/snacks/burger/ppatty/purple
+	name = "purple pretty patty"
+	desc = "Regal and low class at the same time."
+	icon_state = "ppatty-purple"
+	filling_color = "#800080"
+	bitesize = 2
+	list_reagents = list("nutriment" = 7, "protein" = 1)
+	tastes = list("bun" = 2, "meat" = 2, "purple" = 2)
+
+/obj/item/reagent_containers/food/snacks/burger/ppatty/rainbow
+	name = "rainbow pretty patty"
+	desc = "Taste the rainbow, eat the rainbow."
+	icon_state = "ppatty-rainbow"
+	filling_color = "#0000FF"
+	bitesize = 4
+	list_reagents = list("nutriment" = 14, "protein" = 5)
+	tastes = list("bun" = 2, "meat" = 2, "rainbow" = 5)
```

Comment:
The other rainbow thing that I can think of (Cotton candy) has omnizine, you could make that a bit consistent?